Apromore-Core-Components/Apromore-Manager/src/test/java/org/apromore/test/service/impl/EventLogServiceImplTest.java

```diff
@@ -261,4 +294,86 @@ private XLog readXESFile(String fullFilePath) throws Exception {
     return parser.parse(new File(fullFilePath)).get(0);
   }
 
+  @Test
+  public void testGetPerspectiveTagByLog() {
+
+    // Set up test data
+    Integer logId = 1;
+    Log log = createLogWithId(logId, user, createFolder("testFolder", null, wp));
+    Set<Log> logs = new HashSet<>();
+    logs.add(log);
+
+    Usermetadata perspective = createUserMetadata(1, "[\"concept:name\",\"org:resource\",\"lifecycle:transition\"]", logs);
+    Set<Usermetadata> usermetadataSet = new HashSet<>();
+    usermetadataSet.add(perspective);
+
+    // Mock recording
+    expect(userMetadataService.getUserMetadataByLog(logId,
+            UserMetadataTypeEnum.PERSPECTIVE_TAG)).andReturn(usermetadataSet);
+    replayAll();
+
+    // Mock call
+    List<String> perspectives = eventLogService.getPerspectiveTagByLog(logId);
+
+    List<String> result = new ArrayList<>();
+    result.add("concept:name");
+    result.add("org:resource");
+    result.add("lifecycle:transition");
+
+    assertEquals(perspectives, result);
+  }
+
+  @Test
+  public void testGetPerspectiveTagByLog_WithoutMetadata() {
+
+    // Set up test data
+    Integer logId = 1;
+    Log log = createLogWithId(logId, user, createFolder("testFolder", null, wp));
+    Set<Log> logs = new HashSet<>();
+    logs.add(log);
+
+    Usermetadata perspective = createUserMetadata(1, "[\"concept:name\",\"org:resource\",\"lifecycle:transition\"]", logs);
+    Set<Usermetadata> usermetadataSet = new HashSet<>();
+
+    // Mock recording
+    expect(userMetadataService.getUserMetadataByLog(logId,
+            UserMetadataTypeEnum.PERSPECTIVE_TAG)).andReturn(usermetadataSet);
+    replayAll();
+
+    // Mock call
+    List<String> perspectives = eventLogService.getPerspectiveTagByLog(logId);
+
+    List<String> result = new ArrayList<>();
+
+    assertEquals(perspectives, result);
+  }
+
+  @Test
+  public void testGetPerspectiveTagByLog_InvalidJson() {
+
+    // Set up test data
+    Integer logId = 1;
+    Log log = createLogWithId(logId, user, createFolder("testFolder", null, wp));
+    Set<Log> logs = new HashSet<>();
+    logs.add(log);
+    String invalidJSONString = "[concept:name\",\"org:resource\",\"lifecycle:transition\"]";
+    Usermetadata perspective = createUserMetadata(1, invalidJSONString, logs);
+    Set<Usermetadata> usermetadataSet = new HashSet<>();
+    usermetadataSet.add(perspective);
+
+    // Mock recording
+    expect(userMetadataService.getUserMetadataByLog(logId,
+            UserMetadataTypeEnum.PERSPECTIVE_TAG)).andReturn(usermetadataSet);
+    replayAll();
+
+    // Mock call
+    List<String> perspectives = eventLogService.getPerspectiveTagByLog(logId);
+
+    List<String> result = new ArrayList<>();
+
+    assertEquals(perspectives, result);
+    exceptionRule.expectMessage("Could not deserialize JSON content from given JSON content String: " + invalidJSONString);
+    exceptionRule.expect(JsonProcessingException.class);
+  }
+
```

Comment:
Should use @Test(expected = JsonProcessingException.class) to test if eventLogService.getPerspectiveTagByLog throws a JsonProcessingException. The current test is always a success with any exception.